### ISSUE TITLE
android/entityextraction should have unit tests

### ISSUE THREAD

**Issue by ber4444** [googlesamples/mlkit#251]
Please show us how it can be tested. We have a set of textual inputs and a set of entities we currently match in them via regex. We want to switch to MLKit and make sure those same unit tests will keep passing.

**Comment by ber4444** [comment]
it kind of works in robolectric if we do
```
val ctx: Context = ApplicationProvider.getApplicationContext()
        WorkManagerTestInitHelper.initializeTestWorkManager(ctx)
        com.google.mlkit.common.MlKit.initialize(ctx)
```
but when we need to call `downloadModelIfNeeded`,  we can't get it to work... maybe `androidTest` is the only supported solution, sadly?

**Comment by ber4444** [comment]
similar question suggesting to write an androidTest as the only workaround - https://stackoverflow.com/questions/30391563/asynchronous-test-code-in-robolectric-3-with-play-services

**Comment by ber4444** [comment]
yes, it does work in an androidTest but too bad we cannot test it without a device/emulator

**Comment by jackqdyulei** [comment]
Hi Gabor,

I am afraid androidTest is the only choice since EE involves model downloading which is too challenge for robolectric.

In MLKit we do large scale regression test on emulator, where we retrieve test case from dataset, do EE inference and compare result with golden data.

**Comment by ber4444** [comment]
is there a way you could drop in a JVM-only version so we could unit test like this:
`if ("robolectric" != Build.FINGERPRINT) downloadModelsAndroid() else downloadModelsJVM()` etc
I suppose there is nothing in your sdk that couldn't be done on a desktop JVM?
